### PR TITLE
Initial OOP tiny js world commit

### PR DESCRIPTION
**Pull request by yaripey**
### OOP-exercise

[Demo](https://yaripey.github.io/a-tiny-JS-world/)
[Code base](https://github.com/yaripey/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@yaripey your demo leads to a web page based on a different code base.

File: submissions/yaripey/oop-exercise/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+print(dog.aboutMe)
+print(girl.aboutMe)
+print(cat.aboutMe)
+print(boy.aboutMe)
```
Comment (OleksiyRudenko) [inline submissions/yaripey/oop-exercise/index.js:103]: Why such a stepback in iteration through inhabitants?

File: submissions/yaripey/oop-exercise/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+class Inhabitant {
+  constructor(species, name, legs, hands, gender, message) {
+    this.species = species
+    this.name = name
+    this.legs = legs
+    this.gender = gender
+    this.message = message
+    this.friendship = null
+    this.hands = hands
```
Comment (OleksiyRudenko) [inline submissions/yaripey/oop-exercise/index.js:21]: Not every inhabitant has hands by natural design. The beauty of OOP is that every class has props it really needs.
QUick scan didn't show any hands count output

File: submissions/yaripey/oop-exercise/index.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  get aboutMe() {
+    console.log(this.friendship ? true : false)
```
Comment (OleksiyRudenko) [inline submissions/yaripey/oop-exercise/index.js:25]: Debugging artifact? Please remove

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@yaripey please test your app

**Comment by yaripey** [comment]
Links in PR are right, I just messed up with branches a little bit, GitHub pages was pointing on the wrong one. Now it should work correctly.

**Comment by OleksiyRudenko** [comment]
@yaripey what about legs output?
Please feel free asking yourself this type of question with regards to every property individually and task requirements overall.

**Review by OleksiyRudenko — APPROVED** [review]
@yaripey well done. Approved with a reservation.

`.legs` prop is treated in different ways for humans and pets (where those are a substitute for paws). Perhaps these should be different props (`paws` for pets, still `legs` for humans).

Imagine you introduce birds and use `hands` as a substitute for wings. This makes code not really obvious.

**Comment by yaripey** [comment]
Understood, I will consider it in future projects. Thanks!

**Pull request merged**